how-it-works/1_protocol/04_execution.card.md

```diff
@@ -6,10 +6,15 @@ title: Execution Layer
 
 # Execution layer
 
-The execution layer processes one input at a time. This input is taken from one of the
-input queues, and is directed to one canister. Based on this input and the state of the
-canister, the execution environment updates the state of the canister, and additionally may
-add messages to output queues and update the ingress history (possibly with a response to
-an earlier ingress message).
+The execution layer, the topmost layer of the core IC protocol stack, is responsible for executing *queries* (the easy part) and scheduling and executing *canister messages* (the hard part).
```

Comment:
Uhhh....what is a "query"? what is a "canister message". Actually, above in Message Routing we talk about "inter-canister messages" and "canister messages", but never really define them.

General rule: pity the poor reader. What the hell do all these newfangled terms mean? Sorry to rant...